### PR TITLE
main/supervisor: upgrade to 4.0.4

### PR DESCRIPTION
**Pull request by andypost**
Ref http://supervisord.org/changes.html

PS py-meld3 dependency moved to py3

### CODE REVIEW

File: main/supervisor/APKBUILD [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+pkgver=4.0.2
 pkgrel=0
 pkgdesc="A system for controlling process state under UNIX"
 url="http://supervisord.org"
 arch="noarch"
 license="BSD"
-depends="python2 py-meld3 py-setuptools"
-makedepends="$depends_dev"
+depends="python3 py3-meld3"
+makedepends="$depends_dev python3-dev"
```
Comment (TBK) [inline main/supervisor/APKBUILD]:
```suggestion
makedepends="python3-dev"
```

File: main/py3-meld3/APKBUILD [hunk 2]
```diff
@@ -1,41 +1,33 @@
...
 arch="noarch"
 license="ZPL"
-depends="python2"
-depends_dev="python2-dev py-setuptools"
-makedepends="$depends_dev"
-install=""
-subpackages=""
+depends="python3"
+depends_dev=""
+makedepends="$depends_dev python3-dev"
```
Comment (TBK) [inline main/py3-meld3/APKBUILD]:
```suggestion
makedepends="python3-dev"
```

File: main/py3-meld3/APKBUILD [hunk 3]
```diff
@@ -1,41 +1,33 @@
...
 url="https://github.com/supervisor/meld3"
 arch="noarch"
 license="ZPL"
-depends="python2"
-depends_dev="python2-dev py-setuptools"
-makedepends="$depends_dev"
-install=""
-subpackages=""
+depends="python3"
+depends_dev=""
```
Comment (TBK) [inline main/py3-meld3/APKBUILD]:
```suggestion
```

File: main/py3-meld3/APKBUILD [hunk 4]
```diff
@@ -1,41 +1,33 @@
...
-pkgname=py-meld3
+pkgname=py3-meld3
 _pkgname=meld3
 pkgver=1.0.2
-pkgrel=1
+pkgrel=2
 pkgdesc="An HTML/XML templating engine"
 url="https://github.com/supervisor/meld3"
 arch="noarch"
 license="ZPL"
```
Comment (TBK) [inline main/py3-meld3/APKBUILD]: License is incorrect https://github.com/Supervisor/meld3/blob/master/LICENSE.txt

File: main/py3-meld3/APKBUILD [hunk 5]
```diff
@@ -1,41 +1,33 @@
...
-depends_dev="python2-dev py-setuptools"
-makedepends="$depends_dev"
-install=""
-subpackages=""
+depends="python3"
+depends_dev=""
+makedepends="$depends_dev python3-dev"
 source="https://files.pythonhosted.org/packages/source/${_pkgname:0:1}/$_pkgname/$_pkgname-$pkgver.tar.gz"
 
 _builddir="$srcdir"/$_pkgname-$pkgver
```
Comment (TBK) [inline main/py3-meld3/APKBUILD]:
```suggestion
builddir="$srcdir"/$_pkgname-$pkgver
```

File: main/py3-meld3/APKBUILD (hunk 4 above)
Comment (andypost) [reply main/py3-meld3/APKBUILD]: Udated to "Repose" cos bsd-4 is not the same

File: main/supervisor/APKBUILD [hunk 6]
```diff
@@ -1,14 +1,14 @@
 # Contributor: Peter Bui <pnutzh4x0r@gmail.com>
 # Maintainer:
 pkgname=supervisor
-pkgver=3.3.5
+pkgver=4.0.2
 pkgrel=0
 pkgdesc="A system for controlling process state under UNIX"
 url="http://supervisord.org"
 arch="noarch"
 license="BSD"
```
Comment (maxice8) [inline main/supervisor/APKBUILD]: needs SPDX identifier

File: main/supervisor/APKBUILD [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
-depends="python2 py-meld3 py-setuptools"
-makedepends="$depends_dev"
+depends="python3 py3-meld3"
+makedepends="python3-dev"
```
Comment (maxice8) [inline main/supervisor/APKBUILD]: can be py3-setuptools

File: main/supervisor/APKBUILD [hunk 8]
```diff
@@ -1,14 +1,14 @@
...
 source="$pkgname-$pkgver.tar.gz::https://github.com/Supervisor/$pkgname/archive/$pkgver.tar.gz
```
Comment (maxice8) [inline main/supervisor/APKBUILD]: supervisor can have supervisor-openrc subpackage

File: main/py3-meld3/APKBUILD [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+pkgname=py3-meld3
+_pkgname=meld3
+pkgver=1.0.2
+pkgrel=2
+pkgdesc="An HTML/XML templating engine"
+url="https://github.com/supervisor/meld3"
+arch="noarch"
+license="Repoze"
+depends="python3"
+makedepends="python3-dev"
```
Comment (maxice8) [inline main/py3-meld3/APKBUILD]: dependencies for testing can be added as

 checkdepends="py3-pytest py3-pytest-cov"


But requires moving py3-meld3 and supervisor to testing

Comment (tcely) [reply main/py3-meld3/APKBUILD]: @maxice8 
> But requires moving py3-meld3 and supervisor to testing

I'm not sure what you are trying to say here.

**Comment by andypost** [comment]
And it needs maintainer for sure

**Comment by J0WI** [comment]
The example configuration has also changed. Feel free to pcik my merge attempt from [here](https://github.com/alpinelinux/aports/pull/7033/files#diff-76a99b5299cf7701eb1ddb91d46f1c65).

File: main/py3-meld3/APKBUILD (hunk 9 above)
Comment (TBK) [reply main/py3-meld3/APKBUILD]: The dependencies live in testing and for check() to work they would need to be moved to main. An alternative would be to move supervisor and the check depends to community.

Comment (tcely) [reply main/py3-meld3/APKBUILD]: @TBK Thanks, that cleared things up.

**Comment by tcely** [comment]
4.0.3 (2019-05-22) is out.

File: main/supervisor/APKBUILD [hunk 10]
```diff
@@ -1,14 +1,15 @@
 # Contributor: Peter Bui <pnutzh4x0r@gmail.com>
 # Maintainer:
 pkgname=supervisor
-pkgver=3.3.5
+pkgver=4.0.3
 pkgrel=0
 pkgdesc="A system for controlling process state under UNIX"
 url="http://supervisord.org"
 arch="noarch"
 license="BSD"
```
Comment (tcely) [inline main/supervisor/APKBUILD]:
```suggestion
license="custom"
```

This is only inspired by `BSD-4-Clause` but diverged significantly. Please be sure to include both `COPYRIGHT.txt` and `LICENSES.txt` in the package.

File: main/supervisor/APKBUILD (hunk 7 above)
Comment (andypost) [reply main/supervisor/APKBUILD]: Not sure why it may need setup tools? please elaborate

File: main/supervisor/APKBUILD (hunk 10 above)
Comment (andypost) [reply main/supervisor/APKBUILD]: Fixed as suggested

File: main/supervisor/APKBUILD [hunk 11]
```diff
@@ -22,23 +23,24 @@ builddir="$srcdir/$pkgname-$pkgver"
...
 }
 
 package() {
 	cd "$builddir"
-	python2 setup.py install --prefix=/usr --root="$pkgdir" --optimize=1
+	python3 setup.py install --prefix=/usr --root="$pkgdir" --optimize=1
 	install -Dm600 "$srcdir/supervisord.conf" "$pkgdir/etc/supervisord.conf"
 	install -Dm755 "$srcdir/supervisord.initrd" "$pkgdir/etc/init.d/supervisord"
 	install -Dm644 "$srcdir/supervisord.logrotate" "$pkgdir/etc/logrotate.d/supervisord"
+	install -Dm644 COPYRIGHT.txt LICENSES.txt "$pkgdir"/usr/share/licenses/$pkgname
```
Comment (tcely) [inline main/supervisor/APKBUILD]:
```suggestion
	install -Dm644 -t "$pkgdir"/usr/share/licenses/$pkgname COPYRIGHT.txt LICENSES.txt
```

File: main/supervisor/APKBUILD (hunk 7 above)
Comment (tcely) [reply main/supervisor/APKBUILD]: `python3`currently includes setup tools module. Perhaps it should not if we want to continue having `py3-setuptools` as a distinct package?

File: main/py3-meld3/APKBUILD [hunk 12]
```diff
@@ -0,0 +1,32 @@
+# Contributor: Peter Bui <pnutzh4x0r@gmail.com>
+# Maintainer:
+pkgname=py3-meld3
+_pkgname=meld3
+pkgver=1.0.2
+pkgrel=2
+pkgdesc="An HTML/XML templating engine"
+url="https://github.com/supervisor/meld3"
+arch="noarch"
+license="Repoze"
```
Comment (tcely) [inline main/py3-meld3/APKBUILD]: Repoze is not in the license list as far as I can tell. Switch to custom?

Comment (andypost) [reply main/py3-meld3/APKBUILD]: sounds like it should be custom

**Comment by andypost** [comment]
Needs to review licence

File: main/py3-meld3/APKBUILD (hunk 12 above)
Comment (Cogitri) [reply main/py3-meld3/APKBUILD]: Seems like it's just a plain old `BSD-4-Clause`, no?

File: main/supervisor/APKBUILD [hunk 13]
```diff
@@ -1,14 +1,15 @@
...
-pkgver=3.3.5
+pkgver=4.0.4
 pkgrel=0
 pkgdesc="A system for controlling process state under UNIX"
 url="http://supervisord.org"
 arch="noarch"
-license="BSD"
-depends="python2 py-meld3 py-setuptools"
-makedepends="$depends_dev"
+license="custom"
```
Comment (Cogitri) [inline main/supervisor/APKBUILD]: Same here,  I think it's BSD-4-Clause

File: main/py3-meld3/APKBUILD (hunk 12 above)
Comment (SuperSandro2000) [reply main/py3-meld3/APKBUILD]: Seems different. See https://directory.fsf.org/wiki/License:BSD-4-Clause . Especially point 4.

**Comment by J0WI** [comment]
4.1.0 have been released.

Would be great to have this in the next release. This package is often used in Docker containers.

**Comment by Cogitri** [comment]
Since this is stale superseding this sounds like a good idea.

**Pull request closed without merging**